feature-libs/order/core/facade/order-history-extended-view.service.ts

```diff
@@ -0,0 +1,266 @@
+      orderHistoryListRequest,
+      returnRequestListRequest,
+    ]).pipe(
+      switchMap(
+        (
+          responses: [
+            OrderHistoryListView | undefined,
+            ReturnRequestList | undefined
+          ]
+        ) => {
+          const returnRequests = responses?.[1]?.returnRequests;
+          const orderHistory = responses?.[0];
+          if (returnRequests && orderHistory?.orders) {
+            if (orderHistory.pagination?.totalResults === 0) {
+              return of(orderHistory);
+            }
+            return orderHistory.orders.map((order) => {
+              const returnItems = returnRequests?.filter(
+                (returnItem) => returnItem.order?.code === order.code
+              );
+              if (returnItems) {
+                order.returnRequests = returnItems;
+              }
+              return orderHistory;
+            });
+          } else {
+            return of(orderHistory);
+          }
+        }
+      )
+    );
+  }
+
+  private getOrderDetailsValue(code: string): Observable<Order> {
+    return this.store
+      .select(getOrderById(code))
+      .pipe(filter((order) => Boolean(order)));
+  }
+
+  private getOrderDetailsState(
+    code: string
+  ): Observable<StateUtils.LoaderState<Order>> {
+    return this.store.select(getOrderByIdState(code));
+  }
+
+  private loadOrderDetails(code: string) {
+    this.userIdService.takeUserId(true).subscribe({
+      next: (userId) =>
+        this.store.dispatch(
+          new OrderActions.LoadOrderByID({
+            userId,
+            code,
+          })
+        ),
+    });
+  }
+
+  getOrderDetails(code: string): Observable<Order> {
+    const loading$ = this.getOrderDetailsState(code).pipe(
+      auditTime(0),
+      tap((state) => {
+        if (!(state.loading || state.success || state.error)) {
+          this.loadOrderDetails(code);
+        }
+      })
+    );
+    return using(
+      () => loading$.subscribe(),
+      () => this.getOrderDetailsValue(code)
+    );
+  }
+
+  private getConsignmentTrackingValue(
+    orderCode: string,
+    consignmentCode: string
+  ): Observable<ConsignmentTracking> {
+    return this.store
+      .select(getConsignmentTrackingById(orderCode, consignmentCode))
+      .pipe(filter((tracking) => Boolean(tracking)));
+  }
+
+  private getConsignmentTrackingState(
+    orderCode: string,
+    consignmentCode: string
+  ): Observable<StateUtils.LoaderState<ConsignmentTracking>> {
+    return this.store.select(
+      getConsignmentTrackingByIdState(orderCode, consignmentCode)
+    );
+  }
+
+  private loadConsignmentTracking(orderCode: string, consignmentCode: string) {
+    this.userIdService.takeUserId(true).subscribe({
+      next: (userId) =>
+        this.store.dispatch(
+          new OrderActions.LoadConsignmentTrackingByID({
+            orderCode,
+            consignmentCode,
+            userId,
+          })
+        ),
+    });
+  }
+
+  getConsignmentTracking(
```

Comment:
[MEDIUM]
Let's make this method `protected`. it's not used anywhere else

Reply:
shall we leave this method public? as this is like getter method for the new ngrx entity?